cmd/admin-bucket-remote-set.go

```diff
@@ -0,0 +1,214 @@
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET http(s)://ACCESSKEY:SECRETKEY@TARGET_URL/TARGET_BUCKET [--path | --api]
+
+TARGET:
+   Also called as alias/sourcebucketname
+
+TARGET_BUCKET:
+  Also called as remote target bucket.
+
+TARGET_URL:
+  Also called as remote endpoint.
+
+ACCESSKEY:
+  Also called as username.
+
+SECRETKEY:
+  Also called as password.
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Set a new remote replication target 'replicabucket' on https://minio2:9000 for bucket 'srcbucket' on MinIO server.
+     {{.DisableHistory}}
+     {{.Prompt}} {{.HelpName}} myminio/srcbucket https://foobar:foo12345@minio2:9000/replicabucket --type "replica"
+     {{.EnableHistory}}
+`,
+}
+
+// checkAdminBucketRemoteSetSyntax - validate all the passed arguments
+func checkAdminBucketRemoteSetSyntax(ctx *cli.Context) {
+	argsNr := len(ctx.Args())
+	if argsNr < 2 {
+		cli.ShowCommandHelpAndExit(ctx, "set", 1) // last argument is exit code
+	}
+	if argsNr > 2 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args().Tail()...),
+			"Incorrect number of arguments for remote set command.")
+	}
+	if !ctx.IsSet("type") {
+		fatalIf(errInvalidArgument().Trace(ctx.Args()...), "type flag is required")
+	}
+
+}
+
+// RemoteMessage container for content message structure
+type RemoteMessage struct {
+	op           string
+	Status       string `json:"status"`
+	AccessKey    string `json:"accessKey,omitempty"`
+	SecretKey    string `json:"secretKey,omitempty"`
+	SourceBucket string `json:"sourceBucket"`
+	TargetURL    string `json:"TargetURL,omitempty"`
+	TargetBucket string `json:"TargetBucket,omitempty"`
+	RemoteARN    string `json:"RemoteARN,omitempty"`
+	Path         string `json:"path,omitempty"`
+	API          string `json:"api,omitempty"`
+	Type         string `json:"type"`
+}
+
+func (r RemoteMessage) String() string {
+	switch r.op {
+	case "list":
+		message := console.Colorize("TargetURL", fmt.Sprintf("%s ", r.TargetURL))
+		message += console.Colorize("SourceBucket", r.SourceBucket)
+		message += console.Colorize("Arrow", "->")
+		message += console.Colorize("TargetBucket", r.TargetBucket)
+		message += " "
+		message += console.Colorize("ARN", r.RemoteARN)
+		return message
+	case "remove":
+		return console.Colorize("RemoteMessage", "Removed Remote target for `"+r.SourceBucket+"` bucket successfully.")
+	case "set":
+		return console.Colorize("RemoteMessage", "Remote ARN = `"+r.RemoteARN+"`.")
+	}
+	return ""
+}
+
+// JSON returns jsonified message
+func (r RemoteMessage) JSON() string {
+	r.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(r, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// fetchRemoteTarget - returns the dest bucket, dest endpoint, access and secret key
+func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin.BucketTarget, err error) {
```

Comment:
This function should return *probe.Error

```suggestion
func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin.BucketTarget, err *probe.Error) {
```